mava/wrappers/Gigastep.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Dict, Tuple
+
+import jax
+import jax.numpy as jnp
+from chex import Array, PRNGKey
+from gigastep.gigastep_env import GigastepEnv
+from jumanji import specs
+from jumanji.types import StepType, TimeStep, restart
+from jumanji.wrappers import Wrapper
+
+from mava.types import Observation, ObservationGlobalState, State
+
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
```

Comment:
using data class from flax.struct results in 
![image](https://github.com/instadeepai/Mava/assets/91753931/dc215065-7399-48ef-84c0-d47617ec7f05)
when running the pre-commit even if the number of arguments and their types matches the definition.

So i took this workaround from the jaxMarl file which (i assume) imports the dataclass from dataclasses when the type checking is happening otherwise it imports it from flax.struct